From 110d792d68ec8b683db1954875de06f3cd01ec37 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Mon, 14 Nov 2022 11:23:45 +0200
Subject: [PATCH] [SW-110273] gaudi2: add HL225B efuse cfg FIT

Add a target for creating a FIT that marks HL225B in EFUSE.

Change-Id: I161d9895209bc24988cc3bfd41e937b197c87006
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/scripts/make/gaudi2/Makefile b/scripts/make/gaudi2/Makefile
index 7d34817..907abac 100644
--- a/scripts/make/gaudi2/Makefile
+++ b/scripts/make/gaudi2/Makefile
@@ -151,14 +151,15 @@
 	${HL_DEV}-security_enable.itb		\
 	${HL_DEV}-otp.itb				\
 	${HL_DEV}-otp-local.itb		\
-	${HL_DEV}-eeprom-recovery.itb
+	${HL_DEV}-eeprom-recovery.itb	\
+	${HL_DEV}-hl225b_enable.itb
 
 .PHONY: gaudi2-internal-fit-create
 gaudi2-internal-fit-create: gaudi2-nic-en-create	\
 				gaudi2-eeprom-create-default gaudi2-write-tpm-devid-create	\
 				gaudi2-slt-set-create gaudi2-security-enable-create	\
 				gaudi2-otp-create gaudi2-otp-local-create	\
-				gaudi2-eeprom-create-recovery
+				gaudi2-eeprom-create-recovery gaudi2-hl225b-enable-create
 
 	@$(call hl-check-files,$(GAUDI2_INTERNAL_FITS))
 	@tar czf $(IMAGES_DIR)/${HL_DEV}-int-fit.tgz -C $(IMAGES_DIR)/ $(GAUDI2_INTERNAL_FITS)
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index 28c67d0..ba5ed14 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -158,6 +158,11 @@
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
 		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh security_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
 
+.PHONY: %-hl225b-enable-create
+%-hl225b-enable-create:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh hl225b_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
+
 .PHONY: %-security-enable-install
 %-security-enable-install:
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
